lib/plugins/aws/package/compile/events/cloudFront/index.js

```diff
@@ -243,6 +256,28 @@ class AwsCompileCloudFrontEvents {
     });
   }
 
+  compileCloudFrontCachePolicies() {
+    const userConfig = this.serverless.service.provider.cloudFront || {};
+    if (userConfig.cachePolicies) {
+      const Resources = this.serverless.service.provider.compiledCloudFormationTemplate.Resources;
+      for (const [name, cachePolicyConfig] of Object.entries(userConfig.cachePolicies)) {
+        this.cachePolicies.add(name);
+
+        Object.assign(Resources, {
+          [this.provider.naming.getCloudFrontCachePolicyLogicalId(name)]: {
```

Comment:
I see what you mean, however, consider the following:
- httpApi use the same behavior for authorizers. All user defined authorizers are packaged in a config property in `resolveConfiguration` and converted to CloudFormation resources in `compileAuthorizers`, not depending on their actual usage in relevant `httpApi` events.
- it can be quite frustrating for a user defining some configuration and not seeing it deployed (i.e. : "I changed my service file, how come my CloudFormation template is not affected ?")

What do you think about warning the user instead that he actually defined a policy without using it ?
I append a commit to this PR doing just that. Let me know what you think.

Reply:
> What do you think about warning the user instead that he actually defined a policy without using it ?

Yes, I think it's a right call.

Also it might be that temporarily user decided to not use one, and deleting in such case authorizer on AWS side might be highly unwanted.